specification/library-guidelines.md

```diff
@@ -34,35 +34,69 @@ Applications that use third-party libraries that are instrumented with OpenTelem
 In order to enable telemetry the application must take a dependency on the OpenTelemetry SDK, which implements the delivery of the telemetry. The application must also configure exporters so that the SDK knows where and how to deliver the telemetry. The details of how exporters are enabled and configured are language specific.
 
 
-### API and Minimal Implementation
+## API and Minimal Implementation
 
 The API package is a self-sufficient dependency, in the sense that if the end-user application or a third-party library depends only on it and does not plug a full SDK implementation then the application will still build and run without failing, although no telemetry data will be actually delivered to a telemetry backend.
 
 This self-sufficiency is achieved the following way.
 
-The API dependency will contain a minimal implementation of the API. When no other implementation is explicitly included in the application no telemetry data will be collected. Here is what active components will look like in this case:
+The API dependency contains a minimal implementation of the API. When no other implementation is explicitly included in the application no telemetry data will be collected. Here is what active components look like in this case:
 
 ![Minimal Operation Diagram](language-library-minimal.png)
 
 It is important that values returned from this minimal implementation of API are valid and do not require the caller to perform extra checks (e.g. createSpan() method should not fail and should return a valid non-null Span object). The caller should not need to know and worry about the fact that minimal implementation is in effect. This minimizes the boilerplate and error handling in the instrumented code.
 
 It is also important that minimal implementation incurs as little performance penalty as possible, so that third-party frameworks and libraries that are instrumented with OpenTelemetry impose negligible overheads to users of such libraries that do not want to use OpenTelemetry too.
 
-### SDK Implementation
+## SDK Implementation
 
-The SDK implementation is a separate (optional) dependency. When it is plugged in it substitutes the minimal implementation that is included in the API package (exact substitution mechanism is language dependent). Here is what active components will look like in this case:
+The SDK implementation is a separate (optional) dependency. When it is plugged in it substitutes the minimal implementation that is included in the API package (exact substitution mechanism is language dependent).
+
+SDK implements core functionality that is required for translating API calls into telemetry data that is ready for exporting. Here is the non-comprehensive list of core functionality:
+
+- Scoped span operations.
+- Context propagation.
```

Comment:
I think this may be in the default trivial implementation as well. For the moment I would remove this list and the previous sentence "Here is the non-comprehensive list of core functionality:"

Reply:
I agree, will remove.